tests/unit/test_netcdf.py

```diff
@@ -296,3 +296,40 @@ def test_nc_read_failure(scm_data, test_data_path, caplog):
     assert caplog.record_tuples[0][2] == "Failed reading netcdf file: {}".format(
         test_fname
     )
+
+
+@pytest.mark.parametrize(
+    "mdata", ({}, {"test": "value"}, {"test_int": 1, "test_float": 1.234,},)
+)
+def test_nc_with_metadata(test_scm_run, mdata):
+    test_scm_run.metadata = mdata.copy()
+    with tempfile.TemporaryDirectory() as tempdir:
+        out_fname = join(tempdir, "out.nc")
+
+        run_to_nc(test_scm_run, out_fname, dimensions=("scenario",))
+
+        ds = nc.Dataset(out_fname)
+
+        nc_attrs = {a: ds.getncattr(a) for a in ds.ncattrs()}
+        for k, v in mdata.items():
+            assert k in nc_attrs
+            assert nc_attrs[k] == v
+        assert "created_at" in nc_attrs
+
+        df = nc_to_run(test_scm_run.__class__, out_fname)
+
+        for k, v in mdata.items():
+            assert k in df.metadata
+            assert df.metadata[k] == v
+        assert "created_at" in df.metadata
+
+
+@pytest.mark.parametrize("mdata", ({"test_fails": {"something": "else"},},))
```

Comment:
can you add a test of putting a list and an array in here too?

Reply:
Turn's out they are valid nc attributes!